internal/cluster/connection_manager.go

```diff
@@ -324,30 +328,39 @@ func (m *ConnectionManager) handleInitialMembersAdded(e event.Event) {
 }
 
 func (m *ConnectionManager) handleMembersAdded(event event.Event) {
-	m.logger.Trace(func() string { return "ConnectionManager.handleMembersAdded" })
 	// do not add new members in non-smart mode
 	if !m.smartRouting && m.connMap.Len() > 0 {
 		return
 	}
 	if e, ok := event.(*MembersAdded); ok {
+		m.logger.Trace(func() string {
+			return fmt.Sprintf("connectionManager.handleMembersAdded: %v", e.Members)
+		})
 		missingAddrs := m.connMap.FindAddedAddrs(e.Members, m.clusterService)
 		for _, addr := range missingAddrs {
-			if _, err := m.ensureConnection(context.TODO(), addr); err != nil {
+			if _, err := connectMember(context.TODO(), m, addr); err != nil {
```

Comment:
As far as I understand this also registers a membership listener to the connected member. 
We should not listen from all members. Single member delivers them already. In the case of split brain scenerious, we should follow single members view. 
Same goes for here. 
https://github.com/hazelcast/hazelcast-go-client/pull/617/files#diff-8235afecd77026dd0bddf3a2cd47ca4843345b3ddec987b0a5d45c127d14fadcR686

Reply:
Fixed.